clients/fides-js/src/fides.ts

```diff
@@ -85,36 +93,148 @@ declare global {
 // eslint-disable-next-line no-underscore-dangle,@typescript-eslint/naming-convention
 let _Fides: Fides;
 
+/**
+ * Determines if and when to call the API to retrieve geolocation
+ */
+const retrieveEffectiveGeolocation = async (
+  options: FidesOptions,
+  geolocation: UserGeolocation | undefined
+): Promise<UserGeolocation | undefined> => {
+  let effectiveGeolocation;
+  if (!constructFidesRegionString(geolocation)) {
+    if (options.isGeolocationEnabled) {
+      effectiveGeolocation = await getGeolocation(
+        options.geolocationApiUrl,
+        options.debug
+      );
+    } else {
+      debugLog(
+        options.debug,
+        `User location is required but could not be retrieved because geolocation is disabled.`
+      );
+    }
+  }
+  return effectiveGeolocation;
+};
+
 /**
  * Initialize the global Fides object with the given configuration values
  */
-const init = async (config: FidesConfig) => {
+const init = async ({
+  consent,
+  experience,
+  geolocation,
+  options,
+}: FidesConfig) => {
   // Configure the default consent values
   const context = getConsentContext();
   const consentDefaults = makeConsentDefaults({
-    config: config.consent,
+    config: consent,
     context,
   });
 
   // Load any existing user preferences from the browser cookie
   const cookie = getOrMakeFidesCookie(consentDefaults);
 
-  await initOverlay({
-    consentDefaults,
-    experience: config.experience,
-    geolocation: config.geolocation,
-    options: config.options,
-  });
-
   // Initialize the window.Fides object
   _Fides.consent = cookie.consent;
   _Fides.fides_meta = cookie.fides_meta;
   _Fides.identity = cookie.identity;
-  _Fides.experience = config.experience;
-  _Fides.geolocation = config.geolocation;
-  _Fides.options = config.options;
-
+  _Fides.experience = experience;
+  _Fides.geolocation = geolocation;
+  _Fides.options = options;
   _Fides.initialized = true;
+
+  // TODO: generate device id if it doesn't exist
+
+  debugLog(
+    options.debug,
+    "Validating Fides consent overlay options...",
+    options
+  );
+  if (!validateOptions(options)) {
+    debugLog(options.debug, "Invalid overlay options", options);
+    return;
+  }
+
+  let effectiveGeolocation = geolocation;
+  let effectiveExperience = experience;
+
+  if (!effectiveExperience) {
+    // If experience is not provided, we need to retrieve it via the Fides API.
+    // In order to retrieve it, we first need a valid geolocation, which is either provided
+    // OR can be obtained via the Fides API
+    effectiveGeolocation = await retrieveEffectiveGeolocation(
+      options,
+      geolocation
+    );
+    const userLocationString = constructFidesRegionString(effectiveGeolocation);
+    if (!userLocationString) {
+      debugLog(
+        options.debug,
+        `User location could not be constructed from location params`,
+        effectiveGeolocation
+      );
+      return;
```

Comment:
In other words, if `retrieveEffectiveGeolocation` looked like:

```typescript
const retrieveEffectiveGeolocation = async (
  options: FidesOptions,
  geolocation: UserGeolocation | undefined
): Promise<UserGeolocation | undefined> => {
  // Return the user passed in geolocation if it was provided
  if (geolocation) {
    return geolocation;
  }

  // If geolocation is not enabled, return undefined
  if (!options.isGeolocationEnabled) {
    debugLog(
      options.debug,
      `User location is required but could not be retrieved because geolocation is disabled.`
    );
    return undefined;
  }

  // Call the geolocation API
  return await getGeolocation(
    options.geolocationApiUrl,
    options.debug
  );
};
```

then it doesn't have to worry about validation, since it looks like there'll be a validation on the `UserGeolocation` afterwards anyway?

Reply:
oh I see where the confusion is. So I would like to handle this use case:

1. `geolocation` is passed in
2. `geolocation` is invalid
3. We have geolocationApiUrl, so we can attempt to get a valid geolocation  _after_ we check that it's invalid

But you're totally right! This logic can be much simpler and I can move the initial validation check outside this function

Reply:
ohhh thanks for clearing that up 😄